7.0-cli/bin/magento-extension-installer

```diff
@@ -0,0 +1,424 @@
+        }
+
+        return $manifest["name"];
+    }
+
+    /**
+     * Check if Magento 2 is installed in the given directory.
+     *
+     * @param string $directory
+     *
+     * @return bool
+     */
+    protected function isMagentoInstalled($directory)
+    {
+        $it = new \FilesystemIterator($directory);
+
+        return $it->valid();
+    }
+
+    /**
+     * Install a package with Composer.
+     *
+     * Installs in to the current directory, unless the destination directory
+     * is specified.
+     *
+     * If the package source directory is specified, will add it as a symlinked
+     * repository to the project.
+     *
+     * @param string      $package_name
+     * @param string|null $destination_dir The destination Composer project.
+     * @param string|null $source_dir      The source directory for the package.
+     *
+     * @return bool
+     */
+    protected function installComposerPackage($package_name, $destination_dir = null, $source_dir = null)
+    {
+        if ($source_dir) {
+            $repo_config = json_encode([
+                "type"    => "path",
+                "url"     => $source_dir,
+                "options" => [
+                    "symlink" => true,
+                ]
+            ]);
+
+            if (!$this->composer("config repositories.$package_name '$repo_config'", $destination_dir)) {
+                return false;
+            }
+        }
+
+        return $this->composer("require $package_name @dev", $destination_dir);
+    }
+
+    /**
+     * Check if the given symlink file exists.
+     *
+     * @param string $symlink
+     *
+     * @return bool
+     * @throws Exception
+     */
+    protected function checkSymlink($symlink)
+    {
+        if (file_exists($symlink) && !is_link($symlink)) {
+            throw new \Exception(sprintf("File %s exists, but is not a symlink!", $symlink));
+        }
+
+        return file_exists($symlink);
+    }
+
+    /**
+     * Create a symlink.
+     *
+     * @param string $target
+     * @param string $link
+     *
+     * @return bool
+     */
+    protected function createSymlink($target, $link)
+    {
+        @mkdir(dirname($link), 0755, true);
+        return symlink($target, $link);
+    }
+
+    /**
+     * Run Magento 2 cli commands to enable and configure the given module.
+     *
+     * @param string $name
+     *
+     * @return bool
+     */
+    protected function configureModule($name)
+    {
+        return $this->magentoCommand("module:enable $name")
+            && $this->magentoCommand("setup:upgrade")
+            && $this->magentoCommand("setup:di:compile")
+            && $this->magentoCommand("setup:static-content:deploy")
+            && $this->magentoCommand("cache:flush");
+    }
+
+    /**
+     * Run the Magento 2 installer.
+     *
+     * @return bool
+     */
+    protected function magentoInstaller()
```

Comment:
Yeah, it makes sense. Though all of the passthru methods were names after the command they execute, like `composer()` and `magentoCommand()`.